Userland/Libraries/LibWeb/DOM/Element.cpp

```diff
@@ -356,6 +356,32 @@ DOM::ExceptionOr<bool> Element::matches(StringView selectors) const
     return false;
 }
 
+// https://dom.spec.whatwg.org/#dom-element-closest
+DOM::ExceptionOr<DOM::Element const*> Element::closest(StringView selectors) const
+{
+    auto maybe_selectors = parse_selector(CSS::ParsingContext(static_cast<ParentNode&>(const_cast<Element&>(*this))), selectors);
+    if (!maybe_selectors.has_value())
+        return DOM::SyntaxError::create("Failed to parse selector");
+
+    auto matches_selectors = [](CSS::SelectorList selector_list, Element const* element) -> bool {
```

Comment:
This will create a copy of `selector_list` each time it is invoked, let's take the list by const-reference instead.

Reply:
Thanks for the review, I've fixed your comments.

Reply:
The parameter is now const, but not a const-reference - Should be:
```c++
auto matches_selectors = [](CSS::SelectorList const& selector_list, Element const* element) {
```